sound/soc/sof/trace.c

```diff
@@ -58,11 +63,15 @@ static size_t sof_wait_trace_avail(struct snd_sof_dev *sdev,
 	remove_wait_queue(&sdev->trace_sleep, &wait);
 
 out:
+	/* update buffer offset after wait done */
+	host_offset = (sdev->trace_init_offset + sdev->trace_offset) %
+		      buffer_size;
+
 	/* return bytes available for copy */
-	if (sdev->host_offset < pos)
+	if (host_offset < pos)
 		return buffer_size - pos;
 	else
-		return sdev->host_offset - pos;
+		return host_offset - pos;
 }
```

Comment:
I don't like copy/paste code with different comments, can this be a helper function?